coalahtml/_coalahtml/app/controllers/main.js

```diff
@@ -3,50 +3,44 @@
 angular.module('coalaHtmlApp')
   .controller('MainCtrl', function ($scope, $http) {
     var parseCoalaProject = function(projectDir, coalaJSON) {
-      coalaJSON = coalaJSON || projectDir + '/coala.json';
-      $scope.data = [];
-      $http.get(coalaJSON).success(function(response){
-        var knownFiles = {};
-        $scope.content = {};
-
-        var parseResult = function(el){
-          el.affected_code.forEach(function(sourceRange){
-              if (!(sourceRange.file in knownFiles)){
-                  knownFiles[sourceRange.file] = [];
-
-                  $http.get(projectDir + "/" + sourceRange.file)
-                    .success(function(content) {
-                      $scope.content[sourceRange.file] = content.split("\n");
+      $http.get(projectDir + $scope.Constants.file_data)
+        .success(function(file_data) {
+            $scope.file_dict = {};
+            Object.keys(file_data).forEach(function(file_name) {
+              if (!(file_name in $scope.file_dict)) {
+                $scope.file_dict[file_name]=file_data[file_name];
+              }
+            });
```

Comment:
Here you copy file_data into file_dict. Can't you do this with some function or something?